### ISSUE TITLE
Interleave test output.

### ISSUE THREAD

**Issue by dabrahams** [swiftlang/swift-testing#876]
### Description

Sometimes the fastest way to understand what's going on with a program or library is to put some `print` statements in a test.  Unfortunately when I do that the parallel testing causes the output to be scrambled with progress output from other tests running in parallel.  It would be nicer to capture that output in a buffer and dump each buffer to stdout/stderr atomically.

### Swift & OS version (output of `swift --version && uname -a`)

Swift 6

**Comment by grynspan** [comment]
Duplicate of #298

**Comment by dabrahams** [comment]
FWIW, the points of this request and of #298 are very different, and the latter doesn't seem to contain a definitive request or commitment to interleave output. It would even make sense to have that task depend on this one. Indentation is a substantial feature of its own.

**Comment by grynspan** [comment]
Adding indentation isn't the hard part, of course. 🙃

Please review my comments in that issue, in particular [this one[(https://github.com/swiftlang/swift-testing/issues/298#issuecomment-2004008316).

I recognize you're not asking for the exact same thing, but a broad solution to that issue would also be a broad solution to this one, which is why I marked this issue as a duplicate.

**Comment by dabrahams** [comment]
I don’t see a comment from you there and the link itself seems to be a little broken

**Comment by grynspan** [comment]
Link is working for me. Try again.
